sdk/core/core/inc/az_curl_slist.h

```diff
@@ -11,6 +11,12 @@
 
 #include <_az_cfg_prefix.h>
 
+/**
+ * Appends a zero-terminated string to the given CURL list.
+ */
+AZ_NODISCARD az_result
+az_curl_slist_append(struct curl_slist ** const self, char const * const str);
```

Comment:
why do we add this call on top of curl_easy call?

Reply:
@vhvb1989 could you update your branch from master?

Reply:
done

Reply:
There are two reasons:
1. the signature of this function is compatible with actions.
2. it handles CURL errors and returns AZ_RESULT errors.